hail/python/hailtop/aiocloud/aioazure/fs.py

```diff
@@ -530,16 +535,15 @@ async def open(self, url: str) -> ReadableStream:
         parsed_url = self.parse_url(url, error_if_bucket=True)
         if not await self.exists(url):
             raise FileNotFoundError
-        client = self.get_blob_client(parsed_url)
-        return AzureReadableStream(client, url)
+        return AzureReadableStream(self, parsed_url)
 
     @handle_public_access_error
     async def _open_from(self, url: str, start: int, *, length: Optional[int] = None) -> ReadableStream:
         assert length is None or length >= 1
         if not await self.exists(url):
             raise FileNotFoundError
-        client = self.get_blob_client(self.parse_url(url, error_if_bucket=True))
-        return AzureReadableStream(client, url, offset=start, length=length)
+        parsed_url = self.parse_url(url, error_if_bucket=True)
+        return AzureReadableStream(self, parsed_url, offset=start, length=length)
 
     async def create(self, url: str, *, retry_writes: bool = True) -> AsyncContextManager[WritableStream]:  # pylint: disable=unused-argument
         parsed_url = self.parse_url(url, error_if_bucket=True)
```

Comment:
I'm acknowledging for the record that we discussed whether we should recreate the blob client on create/write. We agreed that seems necessary to avoid loss of authorization on long running uploads; however, we were concerned that the blob client class might hold important state about an upload. We decided to revisit this when we started to practically experience these errors.